File: Dec2020/5Dec/SplitAStringInBalancedStrings.cpp
/*
Problem name: 1221. Split a String in Balanced Strings
Problem link: https://leetcode.com/problems/split-a-string-in-balanced-strings/
*/

class Solution {
public:
    int balancedStringSplit(string s) {
        int arr[s.length()];
        int count = 0;
        for(int i=0;i<s.length();++i){
            if(s[i] == 'R')
                arr[i] = 1;
            else
                arr[i]=-1;
        }
        for(int i=1;i<s.length();++i){
            arr[i]+=arr[i-1];
            if(arr[i] == 0)
                count++;
        }
        return count;
    }
};

File: Dec2020/5Dec/SplitAStringInBalancedStrings.java
/*
Problem Name: 1221. Split a String in Balanced Strings
Problem Link: https://leetcode.com/problems/split-a-string-in-balanced-strings/
*/
class Solution {
    public int balancedStringSplit(String s) {
        int cnt = 0, balance = 0;
        for(char c: s.toCharArray()){
            switch(c){
                case 'R': balance++;break;
                case 'L': balance--;break;
            }
            if(balance == 0)
                cnt++;
        }
        return cnt;
    }
}
